extensions/bitwarden/src/hooks.ts

```diff
@@ -0,0 +1,43 @@
+import { getLocalStorageItem, showToast, ToastStyle, removeLocalStorageItem, setLocalStorageItem } from "@raycast/api";
+import { useState, useEffect } from "react";
+import { BitwardenApi } from "./api";
+
+const LocalStorageSessionKey = "sessionToken"
+export function useBitwarden(bitwardenApi: BitwardenApi): [string | null | undefined, (sessionToken: string | null) => void] {
+  const [sessionToken, setSessionToken] = useState<string | null>();
+
+  useEffect(() => {
+    async function getSessionToken() {
+      const sessionToken = await getLocalStorageItem<string>(LocalStorageSessionKey);
+
+      const status = await bitwardenApi.getVaultStatus(sessionToken)
+
+      if (status === "unlocked") setSessionToken(sessionToken);
+      else if (status === "locked") setSessionToken(null);
+      else if (status === "unauthenticated") {
```

Comment:
Let's use a `switch`/`case` statement to clean up this conditional. Separately, it's an error-prone practice to omit the `{ }` in `if` statements, so if you'd rather not use a `switch`/`case`, at least add those to the first two `if` statements here.

WRT the behavior in each codepath - why are we setting the `sessionToken` in the first two cases? Can't we just do nothing? I would expect something like:
```ts
switch(status) {
  case "unlocked":
    return;
  case "locked":
  case "unauthenticated":
    // Prompt for login
}
```

Reply:
It is related to the react logic. The setSessionToken triggers a rerender of the component, allowing to either :
- Render the form (if sessionToken is set to null)
- Render the list of items

I agree that it can feel a bit strange if you are not familiar with react, but this was the best way i found to implement this logic.